### PR TITLE
Add react hooks `use-state` and `use-memo`

### PR DESCRIPTION
**Pull request by clauxx**
fixes #19016

### Summary

Adding react hooks: `use-state` and `use-memo`.

status: ready

### CODE REVIEW

**Review by flexsurfer — APPROVED** [review]
yes, thank you, it should have been merged in my PR, but because of e2e it's delayed

**Comment by flexsurfer** [comment]
@clauxx it also needs these two files, if you want to use state
https://github.com/status-im/status-mobile/pull/18905/files#diff-5a390ad854e6b64f7c0310ba3edda062d511bf9dbf1314fa26116052efbe997f
https://github.com/status-im/status-mobile/pull/18905/files#diff-51b9719d71ee66feae404b7601830204b4c7af1798bb16a9089d466fccfcb84a

**Review by flexsurfer — CHANGES_REQUESTED** [review]
more changes needed

**Review by flexsurfer — APPROVED** [review]

**Comment by clauxx** [comment]
@pavloburykh as discussed, this PR doesn't change any current functionality and it's needed by a few issues we're working on atm. Since E2E tests are currently broken, it'll be merged without E2E tests.

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/navigation/view.cljs [hunk 1]
```diff
@@ -17,6 +17,8 @@
     [status-im.setup.hot-reload :as reloader]
     [utils.re-frame :as rf]))
 
+(def functional-compiler (reagent/create-compiler {:function-components true}))
```
Comment (J-Son89) [inline src/status_im/navigation/view.cljs:20]: it's a bit weird to put the compiler related code in the navigation related code.

**Comment by J-Son89** [comment]
looking at this code, it's not clear that it should have skipped manual QA. It claims to be just adding two functions not being used yet but you're adding code to the core navigation which can easily break/ introduce subtle bugs.

File: src/status_im/navigation/view.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/navigation/view.cljs:20]: it's related to reagent, and we use reagent here directly so its fine

Comment (flexsurfer) [reply src/status_im/navigation/view.cljs:20]: for some reason it didn't work correctly with hooks by just setting default compiler

**Comment by flexsurfer** [comment]
> looking at this code, it's not clear that it should have skipped manual QA. It claims to be just adding two functions not being used yet but you're adding code to the core navigation which can easily break/ introduce subtle bugs.

it was tested in another PR, it doesn't affect old code, only code with hooks which is not presented

**Pull request merged**